assets/app/scripts/controllers/edit/buildConfig.js

```diff
@@ -0,0 +1,617 @@
+              $scope.options.pickedImageSourceDockerImage = (imageSourceFrom.kind === "ImageStreamTag") ? buildConfig.metadata.namespace + "/" + imageSourceFrom.name : imageSourceFrom.name;
+
+              $scope.imageSourceImageStream = {
+                namespace: $scope.options.pickedImageSourceNamespace,
+                imageStream: $scope.options.pickedImageSourceImageStream,
+                tag: $scope.options.pickedImageSourceImageStreamTag,
+              };
+            }
+
+            if ($scope.strategyType === "Docker") {
+              $scope.options.noCache = !!$scope.buildConfig.spec.strategy.dockerStrategy.noCache;
+              // Only DockerStrategy can have empty Strategy object and therefore it's from object
+              $scope.buildFromTypes.push({"id": "None", "title": "--- None ---"});
+            }
+
+            $scope.buildFrom.projects = ["openshift"];
+            DataService.list("projects", $scope, function(projects) {
+              var projects = projects.by("metadata.name");
+              for (var name in projects) {
+                $scope.buildFrom.projects.push(name);
+                $scope.pushTo.projects.push(name);
+              }
+
+              // If builder or output image reference kind is DockerImage select the first imageSteam and imageStreamTag
+              // in the picker, so when the user changes the reference kind to ImageStreamTag the picker is filled with
+              // default(first) value.
+              var builderSelectFirstOption = ($scope.options.pickedBuildFromType === "DockerImage") ? true : false;
+              var outputSelectFirstOption = ($scope.options.pickedPushToType === "DockerImage") ? true : false;
+              $scope.updateBuilderImageStreams($scope.options.pickedBuildFromNamespace, builderSelectFirstOption);
+              $scope.updateOutputImageStreams($scope.options.pickedPushToNamespace, outputSelectFirstOption);
+
+              if ($scope.sources.image) {
+                $scope.imageSourceBuildFrom.projects = angular.copy($scope.buildFrom.projects);
+                var imageSourceSelectFirstOption = ($scope.options.pickedImageSourceType === "DockerImage") ? true : false;
+                $scope.updateImageSourceImageStreams($scope.options.pickedImageSourceNamespace, imageSourceSelectFirstOption);
+              }
+
+              angular.element(document).ready(function () {
+                // $('.combobox').combobox();
+              });
+            });
+            $scope.loaded = true;
+            // If we found the item successfully, watch for changes on it
+            watches.push(DataService.watchObject("buildconfigs", $routeParams.buildconfig, context, function(buildConfig, action) {
+              if (action === "DELETED") {
+                $scope.alerts["deleted"] = {
+                  type: "warning",
+                  message: "This build configuration has been deleted."
+                };
```

Comment:
Then it won't let you edit it at all, and redirects you to the bc browse page with a error notification